code.gs

```diff
@@ -380,6 +424,106 @@ function monthToInt (month) {
 }
 
 /*
+ * An object representing a simplified semantic version number.
+ * It must be composed of:
+ *  - three dot-separated positive integers (major version,
+ *    minor version and patch number);
+ *  - optionally a pre-release identifier, prefixed by a hyphen;
+ *  - optionally a metadata identifier, prefixed by a plus sign;
+ * This differes from the official SemVer style because the pre-release
+ * string is compared as a whole in version comparison instead of
+ * being spliced into chuncks.
+ * Valid examples:
+ *  4.6.2, 3.12.234-alpha,  0.11.0+20170827, 2.0.0-beta+20170827
+ */
+function SimplifiedSemanticVersion (versionNumber) {
+  var matches, self;
+
+  self = this;
+  this.numbers = [0, 0, 0];
+  this.prerelease = '';
+  this.metadata = '';
+
+  matches = versionNumber.match(/^(\d+)\.(\d+)\.(\d+)(?:-(.+?))??(?:\+(.+))??$/);
```

Comment:
In the regex, as far as I can see the "lazy match" `??` on both the trailing subscripts is not necessary as the matching is working left-to-right anyway so (unless I'm missing something) perhaps it is unnecessary complexity/fragility (and I suspect the lazy-matching is actually less optimal, but as it is only micro-differences in terms of speed that is not so important). This seems to work just as well for me `/^(\d+)\.(\d+)\.(\d+)(?:-(.+?))?(?:\+(.+?))?$/`. If you copy-pasted that from some "reference implementation" for semantic-versioning can you tell me the link because I'm curious why they do it (I know about the actual [semver spec](https://semver.org), am more interested in the regex implementation for it). Perhaps some regex implementations work right-to-left for subscripts(?!), or maybe the regex-spec just doesn't state it explicitly(?!)...

FWIW: I just looked through [node-semver](https://github.com/npm/node-semver/blob/master/semver.js) (which is of course way too much overkill for what you need), but they don't use `??` for matching either...